### PR TITLE
Add unit test for Score saving flow

### LINKED ISSUE #58: Add tests for Score saving flow
Currently the user flow

**Completing the game -> Render result page -> Enter username -> Save username and score to localStorage.**

is not tested. Using jest and @testing-library/react tests have to be added for this flow.

Test should go in src/tests/Game.test.js

### PR DESCRIPTION
**Pull request by devTeaa**
### Pull Request description

Fixes #58, sorry @itaditya the old PR got nuked accidentally by me

### Checklist for proper Pull Request

* [X] performed `npm run format` on the code.
* [X] check all tests are working.
* [X] [allowed edits from maintainers](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

### CODE REVIEW

File: src/tests/Game.test.js [hunk 1]
```diff
@@ -190,4 +196,26 @@ describe('Move counter', () => {
...
+    const board = getByTestId('game-table');
+
+    fireEvent.keyDown(board, {
+      key: 'ArrowLeft'
+    });
+    fireEvent.keyDown(board, {
+      key: 'ArrowRight'
+    });
+
+    const windowLocalStorage = global.window.localStorage;
```
Comment (itaditya) [inline src/tests/Game.test.js]: I think this is not needed now

**Comment by itaditya** [comment]
fix the merge conflicts and the comment I left

**Comment by devTeaa** [comment]
@itaditya I have fixed as per comment above

**Comment by devTeaa** [comment]
@itaditya seems like there is syntax errors on current master branch, can you check it out?

File: src/tests/Game.test.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { render, fireEvent, cleanup } from '@testing-library/react';
+import { render, fireEvent, cleanup } from 'react-testing-library';
```
Comment (itaditya) [inline src/tests/Game.test.js]:
```suggestion
import { render, fireEvent, cleanup } from '@testing-library/react';
```

**Comment by itaditya** [comment]
yeah there is some issue, but for now I'm just checking if your added test cases passes. Currently you are using react-testing-library which is outdated, now you need to use @testing-library/react.

**Comment by devTeaa** [comment]
`it('should save the score to local storage', () => {
  ...
})`

`it('should have 10 sprite tiles at the begining of the game', () => {
  ...
})`

both of these test throw random error on Game.test.js, any idea why?

Also how to trigger componentDidMount() with @testing-library/react?

**Pull request closed without merging**